File: Herencia/ejercicio 2/Main.java
package herencia;
public class Main {
    public static void main(String[] args) {
        Animal[] zoologico = new Animal[3];

        zoologico[0] = new Leon("Simba", 5);
        zoologico[1] = new Pinguino("Pingu", 3);
        zoologico[2] = new Canguro("Kangoo", 6);

        for (Animal a : zoologico) {
            a.desplazarse();
        }
    }
}

File: Herencia/ejercicio 2/main.py
from leon import Leon
from pinguino import Pinguino
from canguro import Canguro

def main():
    zoologico = [
        Leon("Simba", 5),
        Pinguino("Pingu", 3),
        Canguro("Kangoo", 6)
    ]

    for animal in zoologico:
        print(animal)
        animal.desplazarse()

if __name__ == "_main_":
    main()